functions/Test-DbaRepLatency.ps1

```diff
@@ -0,0 +1,230 @@
+
+    .PARAMETER SqlInstance
+    Allows you to specify a comma separated list of servers to query.
+
+    .PARAMETER Database
+    The database(s) to process. If unspecified, all databases will be processed.
+
+    .PARAMETER SqlCredential
+    Login to the target instance using alternative credentials. Windows and SQL Authentication supported. Accepts credential objects (Get-Credential)
+
+    .PARAMETER PublicationName
+    The publiction(s) to process. If unspecified, all publications will be processed.
+
+    .PARAMETER TimeToLive
+    How long, in seconds, to wait for a tracer token to complete its journey from the publisher to the subscriber.
+    If unspecified, all tracer tokens will take as long as they need to process results.
+
+    .PARAMETER RetainToken
+    Retains the tracer tokens created for each publication. If unspecified, all tracer tokens created will be discarded.
+
+    .PARAMETER DisplayTokenHistory
+    Displays all tracer tokens in each publication. If unspecified, the current tracer token created will be only token displayed.  
+
+    .PARAMETER EnableException
+    By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+    This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+    Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+    Author: Colin Douglas
+    Tags: Replication
+
+    Website: https://dbatools.io
+    Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+    License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+    https://dbatools.io/Test-DbaRepLatency
+
+    .EXAMPLE
+    Test-DbaRepLatency -SqlInstance sql2008, sqlserver2012
+    Return replication latency for all transactional publications for servers sql2008 and sqlserver2012.
+
+    .EXAMPLE
+    Test-DbaRepLatency -SqlInstance sql2008 -Database TestDB
+    Return replication latency for all transactional publications on server sql2008 for only the TestDB database
+
+    .EXAMPLE
+    Test-DbaRepLatency -SqlInstance sql2008 -Database TestDB -PublicationName TestDB_Pub
+    Return replication latency for the TestDB_Pub publication for the TestDB databasse located on the server sql2008.
+#>
+    [CmdletBinding()]
+    Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [DbaInstanceParameter[]] $SqlInstance, #Publisher
+        [object[]]$Database,
+        [PSCredential]$SqlCredential,
+        [object[]]$PublicationName,
+        [int]$TimeToLive,
+        [switch]$RetainToken,
+        [switch]$DisplayTokenHistory,
+        [switch]$EnableException
+    )
+
+    Begin{
+
+        if ($null -eq [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SqlServer.RMO")) {
+            Write-Host "Replication management objects not available. Please install SQL Server Management Studio."
```

Comment:
This would be a `Stop-Function`, which would stop the command from running. You can see samples within the rest of the command.